### ISSUE TITLE
Is there a way to run h2load for a specified duration? 

### ISSUE THREAD

**Issue by naveen-venkatk** [nghttp2/nghttp2#463]
For example i want to keep running a constant load of 1000 requests for 10 hours at a specific rate

**Comment by shukitchan** [comment]
In the simplest way, I use -c and -r options together to control the duration of the load test. e.g. 

-c 100 -r 10 

should run for roughly 10 seconds, firing 10 connections every second

But of course if you need more sophisticated way of controlling the duration, you will need some changes in the program

**Comment by nvenkat1982** [comment]
Thank you. That should help me keep going for now.

**Comment by tatsuhiro-t** [comment]
If I understand this issue correctly, you are looking for the method wrk uses.
I have considered to add this kind of thing.  The internal of h2load currently depends on the total number of requests, so there should be something should be done to support this, but if it has popularity, I'll consider to add it.
Using -r option may make more realistic load, since it creates more connect/disconnect, and some burst of requests.

**Comment by nvenkat1982** [comment]
That is right. I am looking for someway to run the tests for specified duration like wrk does and bench marking the servers resources. For now as pointed above, i can arrive at duration by using a combination of -r and -C , but having inbuilt support for duration would take the guess work out of it. 
Also if you consider to add that support , it would be great to provide an option where the load keeps running indefinitely  making connections at specified -r till the user decides to stop the load.  

Thanks

**Comment by LPardue** [comment]
A slightly different approach that uses h2load without any modification could be to define a timing script file for your test period. The script would issue requests at intervals that roughly match to the request time of the resource, meaning that there is a consistent load. 

The script this could be generated in a spreadsheet that has autofill features, and exported to a tab-separated file.

However, for a 10 hour test period I would anticipate that there would be a lot of requests, so you may hit resource limitations.

**Comment by superzelda** [comment]
1. there is a input-file lines limitation? it allowed to send 10,000 different urls but not support 50,000 different urls?
2. does the input-file support multiple headers? such as User-Agent, Cookies, etc.

**Comment by tatsuhiro-t** [comment]
@superzelda is your question related to this issue?

**Comment by tatsuhiro-t** [comment]
To answer the question:

> 1. there is a input-file lines limitation? it allowed to send 10,000 different urls but not support 50,000 different urls?

No limitation I'm aware of.  You need the number of requests more than or equals to the URIs in input-file to use them all.  Do you get any error while reading the file?

> 2. does the input-file support multiple headers? such as User-Agent, Cookies, etc.

No support to specify header fields in input-file at the moment.

**Comment by superzelda** [comment]
sorry, my question is not relate to this issue. and thanks for your reply.

**Comment by tatsuhiro-t** [comment]
Time-based test has already been implemented:
https://nghttp2.org/documentation/h2load-howto.html#timing-based-load-testing
